test/intern/mixins/Tree.js

```diff
@@ -409,6 +409,34 @@ define([
 					}, null, 'Modification of child should not throw error');
 				});
 			});
+
+			test.test('child modification after parent when expanded', function() {
+				var dfd = this.async(10000);
+				expand(0).then(function() {
+					grid.collection.putSync({
+						id: '0',
+						value: 'Modified'
+					});
+					grid.collection.putSync({
+						id: '0:0',
+						value: 'Modified',
+						parent: '0'
+					});
+					setTimeout(dfd.callback(function() {
```

Comment:
I removed the `setTimeout` and it still works fine, and still fails(in chrome and IE) after removing the fix. I also removed the fourth call to `putSync` as that's not necessary either. Firefox is stumping me at the moment. Manually reproducing the steps from the original ticket without the fix results in the same broken behavior as in chrome, and it also works correctly after adding the fix. Despite that I can't get it to fail in this test case.